src/kata6.joinNames.js

```diff
@@ -1,3 +1,19 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+    // here I'm using the optional index and source array parameters
+    const resultsArr = namesObj.reduce((accName, currVal, idx, srcArr) => {
```

Comment:
Nice use of the reduce method! Just one small thing, `resultsArr` is actually a string. It's the string passed as the starting value on line 13 but with all the concatenations carried out within the reduce callback. No need to return `String(resultsArr)` on line 16, just return rename `resultsArr` to `result` and return that.